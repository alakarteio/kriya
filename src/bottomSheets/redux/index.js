import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
