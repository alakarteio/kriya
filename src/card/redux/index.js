export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
