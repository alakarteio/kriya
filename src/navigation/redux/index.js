export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
