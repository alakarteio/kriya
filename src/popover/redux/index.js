export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
