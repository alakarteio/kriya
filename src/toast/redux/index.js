import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

