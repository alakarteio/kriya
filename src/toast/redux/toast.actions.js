export const CREATE_TOAST = 'CREATE_TOAST'

const action = type =>
  (title, options) =>
  ({ type: CREATE_TOAST, payload: { title, type, ...options } })

export const create = (title, options) => action()(title, options)
const types = ['success', 'warning', 'error']
types.forEach((type) => { create[type] = action(type) })

export const REMOVE_TOAST = 'REMOVE_TOAST'
export const remove = () => ({ type: REMOVE_TOAST })
