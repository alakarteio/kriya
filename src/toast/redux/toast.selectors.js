export const getCurrentToast = state => state.ui.toast[0]
