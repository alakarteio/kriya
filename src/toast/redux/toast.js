import { CREATE_TOAST, REMOVE_TOAST } from './toast.actions'

export const initState = {}
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case CREATE_TOAST:
      return { ...payload, print: true }
    case REMOVE_TOAST:
      return initState
    default: return state
  }
}
