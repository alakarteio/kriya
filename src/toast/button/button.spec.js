/* eslint-disable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
/* eslint-env jest */
import React from 'react'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import snap from 'snap'
import Button from './button'
import ButtonContainer from './button.container'
import { remove } from '../redux'

const snapshot = props => snap(Button)({ children: [], ...props })

const createContainer = (store, props) => {
  return mount(
    <Provider store={store}>
      <ButtonContainer {...props} />
    </Provider>,
  )
}

describe('common/Button', () => {
  describe('graphical (JSX)', () => {
    it('should have a default behaviour', snapshot({}))
    it('should add custom className', snapshot({ className: 'custom' }))
    it('should add custom style', snapshot({ style: { color: 'blue' } }))
    it('should add text', snapshot({ text: 'my button' }))
    it('should add handler', snapshot({ onClick: () => 'ok' }))
    it('should add custom type', snapshot({ type: 'warning' }))
  })
  describe('container', () => {
    it('should call onClick and remove when button is pressed', () => {
      const dispatch = jest.fn()
      const onClick = jest.fn()
      const state = { ui: { toast: { code: 'toast1', print: true, title: 'toast', button: { text: 'button', onClick } } } }
      const store = createStore(() => state)
      store.dispatch = dispatch
      const component = createContainer(store, state.ui.toast.button)
      component.find('button').first().simulate('click')
      expect(dispatch.mock.calls.length).toBe(1)
      expect(onClick.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([remove()])
    })
  })
})
/* eslint-enable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
