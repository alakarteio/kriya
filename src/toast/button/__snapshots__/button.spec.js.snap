// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Button graphical (JSX) should add custom className 1`] = `null`;

exports[`common/Button graphical (JSX) should add custom style 1`] = `null`;

exports[`common/Button graphical (JSX) should add custom type 1`] = `null`;

exports[`common/Button graphical (JSX) should add handler 1`] = `null`;

exports[`common/Button graphical (JSX) should add text 1`] = `
<button
  className="button"
  onClick={undefined}
  style={Object {}}
>
  my button
</button>
`;

exports[`common/Button graphical (JSX) should have a default behaviour 1`] = `null`;
